allennlp/modules/token_embedders/embedding.py

```diff
@@ -181,12 +181,19 @@ def extend_vocab(self,  # pylint: disable=arguments-differ
         """
         # Caveat: For allennlp v0.8.1 and below, we weren't storing vocab_namespace as an attribute,
         # knowing which is necessary at time of embedding vocab extension. So old archive models are
-        # currently unextendable unless the user used default vocab_namespace 'tokens' for it.
+        # currently unextendable.
 
         vocab_namespace = vocab_namespace or self._vocab_namespace
         if not vocab_namespace:
-            vocab_namespace = "tokens"
-            logging.warning("No vocab_namespace provided to Embedder.extend_vocab. Defaulting to 'tokens'.")
+            # It's not safe to default to 'tokens' when we aren't sure that 'tokens'
+            # need to be extended. (Without this, several tests fail.)
+            logging.warning("No vocab_namespace provided to Embedder.extend_vocab. Extension will be no-op'.")
```

Comment:
In what circumstances will this actually emit a warning?  Will almost everyone that loads or trains a model in practice see this warning?  If so, it should be at the `info` level (or even `debug`).

Reply:
The warning won't be seen for any models trained after #2374 because `_vocab_namespace` is stored. For previously trained models, it would almost always be seen because `extend_vocab` call is implicit now. Will change it to `info` level.